api/capi/src/nnstreamer-capi-util.c

```diff
@@ -550,15 +551,19 @@ ml_tensors_data_destroy (ml_tensors_data_h data)
 
   _data = (ml_tensors_data_s *) data;
 
-  for (i = 0; i < ML_TENSOR_SIZE_LIMIT; i++) {
-    if (_data->tensors[i].tensor) {
-      g_free (_data->tensors[i].tensor);
-      _data->tensors[i].tensor = NULL;
+  if (_data->handle != NULL) {
+    status = ml_single_destroy_notify (_data->handle, _data);
+  } else {
+    for (i = 0; i < ML_TENSOR_SIZE_LIMIT; i++) {
+      if (_data->tensors[i].tensor) {
+        g_free (_data->tensors[i].tensor);
+        _data->tensors[i].tensor = NULL;
```

Comment:
This line is unnecessary. g_free does already set _data->tensors[i].tensor to NULL.

Reply:
I am not sure - https://github.com/bratsche/glib/blob/master/glib/gmem.c#L258